### DISCUSSION TITLE
Displaying Non English characters using DMD library like Hindi, Telugu etc..

### DISCUSSION THREAD

**Discussion by dileeppulluru** [board707/DMD_STM32#32] [General]
Hello all,
             I have P10 RGB modules of 32X16 .My requirement is to arrange 16 modules of p10 in 2 rows and 8 columns and to display information wirelessly through MQTT . I have used ESP32 Matrix hub panel library where it is working properly but I also want to display non English characters like Indian languages namely Hindi, Telugu , etc.. where library doesn't have support.
         I have seen that this library has Unicode support, whether anyone help me regarding this how to approach to this library for chaining of 16 modules as well as support for non English characters.
Am new to STM controller I don't have any idea about it , anyone please explain whether can we get the above requirement or not with STM32  ,if not what are the other options to do?
 
Thanks in advance
@board707

**Comment by board707** [comment]
> what are the other options to do?

Besides STM32 you can also use arduino boards based RP2040 controller. Maybe it will be more convenient than STM32.

To add support for new language, you first need an adafruit font fot this language.
I don't know Hindi, can you clarify if this is a language where each letter has a fixed face or does it change depending on the preceding and following letters in a word, such as in Arabic?

**Reply by dileeppulluru** [reply]
yes it is not a fixed language like English. In English only there are 26 letters but in hindi and telugu for each letter there will be different forms it depends on the preceding and following letter..

Does DMD library supports RP2040??

**Reply by board707** [reply]
> in hindi and telugu for each letter there will be different forms it depends on the preceding and following letter..

I'm afraid this is a very difficult task. In addition to English, I know Russian and Spanish, but all of these languages use fixed letters, like in English. As I do not know Hindi, it will be difficult for me to help you.

> Does DMD library supports RP2040??

Yes, as stated in the page of the library https://github.com/board707/DMD_STM32/tree/dev-V2#compatible-boards

**Comment by board707** [comment]
For the sake of curiosity, tell me how computer fonts for Hindi are arranged - do they contain many spellings for each letter? How many variants are there in the whole language?
